### PR TITLE
Better error messages when trying to use virtual fields in queries

### PR DESCRIPTION
**Pull request by rizasal**
Following this [discussion](https://groups.google.com/g/elixir-ecto/c/pqh9MbEQpcQ)

### CODE REVIEW

**Comment by yordis** [comment]
@rizasal 💘 thank you so much!

File: lib/ecto/query/planner.ex [hunk 1]
```diff
@@ -1732,7 +1732,13 @@ defmodule Ecto.Query.Planner do
         type
 
       Map.has_key?(schema.__struct__(), field) ->
-        error! query, expr, "field `#{field}` in `#{kind}` is a virtual field in schema #{inspect schema}"
+        case schema.__schema__(:association, field) do
+          %Ecto.Association.BelongsTo{owner_key: owner_key} ->
```
Comment (v0idpwn) [inline lib/ecto/query/planner.ex:1736]: What about other types of association? Maybe we could give a specific message for them?

**Review by v0idpwn — COMMENTED** [review]
Hi! Thanks for the PR. It looks good.

File: lib/ecto/query/planner.ex [hunk 2]
```diff
@@ -1732,7 +1732,13 @@ defmodule Ecto.Query.Planner do
...
+            error! query, expr, "field `#{field}` in `#{kind}` is a virtual field in schema #{inspect schema}. " <>
```
Comment (v0idpwn) [inline lib/ecto/query/planner.ex:1737]:
```suggestion
            error! query, expr, "field `#{field}` in `#{kind}` is an association in schema #{inspect schema}. " <>
```

File: lib/ecto/query/planner.ex [hunk 3]
```diff
@@ -1732,7 +1732,13 @@ defmodule Ecto.Query.Planner do
         type
 
       Map.has_key?(schema.__struct__(), field) ->
-        error! query, expr, "field `#{field}` in `#{kind}` is a virtual field in schema #{inspect schema}"
+        case schema.__schema__(:association, field) do
+          %Ecto.Association.BelongsTo{owner_key: owner_key} ->
+            error! query, expr, "field `#{field}` in `#{kind}` is an association in schema #{inspect schema}. " <>
+                                "Did you mean to use :#{owner_key}?"
+          _ ->
+            error! query, expr, "field `#{field}` in `#{kind}` is a virtual field in schema #{inspect schema}"
```
Comment (josevalim) [inline lib/ecto/query/planner.ex:1740]:
```suggestion
            error! query, expr, "field `#{field}` in `#{kind}` is a virtual field or association in schema #{inspect schema}"
```

File: lib/ecto/query/planner.ex [hunk 4]
```diff
@@ -1732,7 +1732,13 @@ defmodule Ecto.Query.Planner do
         type
 
       Map.has_key?(schema.__struct__(), field) ->
-        error! query, expr, "field `#{field}` in `#{kind}` is a virtual field in schema #{inspect schema}"
+        case schema.__schema__(:association, field) do
+          %Ecto.Association.BelongsTo{owner_key: owner_key} ->
+            error! query, expr, "field `#{field}` in `#{kind}` is an association in schema #{inspect schema}. " <>
+                                "Did you mean to use :#{owner_key}?"
```
Comment (josevalim) [inline lib/ecto/query/planner.ex:1738]:
```suggestion
                                "Did you mean to use `#{owner_key}`?"
```

File: test/ecto/query/planner_test.exs [hunk 5]
```diff
@@ -1008,6 +1008,13 @@ defmodule Ecto.Query.PlannerTest do
       query = from(Comment, []) |> select([c], c.temp)
       normalize(query)
     end
+
+    message =
+      ~r"field `crazy_post_with_list` in `select` is a virtual field in schema Ecto.Query.PlannerTest.Comment. Did you mean to use :crazy_post_id?"
```
Comment (josevalim) [inline test/ecto/query/planner_test.exs:1013]:
```suggestion
      ~r"field `crazy_post_with_list` in `select` is a virtual field in schema Ecto.Query.PlannerTest.Comment. Did you mean to use `crazy_post_id`"
```

File: lib/ecto/query/planner.ex (hunk 3 above)
Comment (josevalim) [inline lib/ecto/query/planner.ex:1740]:
```suggestion
          %_{} ->
            error! query, expr, "field `#{field}` in `#{kind}` is an association in schema #{inspect schema}"

          _ ->
            error! query, expr, "field `#{field}` in `#{kind}` is a virtual field in schema #{inspect schema}"
```

File: test/ecto/query/planner_test.exs [hunk 6]
```diff
@@ -1008,6 +1008,13 @@ defmodule Ecto.Query.PlannerTest do
       query = from(Comment, []) |> select([c], c.temp)
       normalize(query)
     end
+
+    message =
+      ~r"field `crazy_post_with_list` in `select` is a virtual field in schema Ecto.Query.PlannerTest.Comment. Did you mean to use `crazy_post_id`"
```
Comment (josevalim) [inline test/ecto/query/planner_test.exs:1013]:
```suggestion
      ~r"field `crazy_post_with_list` in `select` is an association in schema Ecto.Query.PlannerTest.Comment. Did you mean to use `crazy_post_id`"
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**